RFS/src/main/java/com/rfs/common/LuceneDocumentsReader.java

```diff
@@ -36,6 +39,7 @@ public Flux<Document> readDocuments(Path luceneFilesBasePath, String indexName,
                     reader.close();
                 } catch (IOException e) {
                     logger.error("Failed to close IndexReader", e);
```

Comment:
you can also write this as logger.atError().setCause(e).setMessage(()->"...").log().
From my read, when I've looked at this in the past, the multi-arg error function is a pretty loose convention that just happens to work with log4j2, but it isn't really an slf4j standard.

Reply:
I'll do logging pass a bit later, want to get the core functionality in place first.